src/client.js

```diff
@@ -72,12 +72,12 @@ const Client = class Client {
 
   _handleMessage(topic, message) {
     var parts = topic.split('/');
-    if (parts[3] !== 'activations' && parts[3] !== 'up') {
+    if (parts[4] !== 'activations' && parts[3] !== 'up') {
       return;
     }
     var devId = parts[2];
     var payload = JSON.parse(message.toString());
-    if (parts.length === 4) { // inject app_id and dev_id, but not for field topics
+    if (parts[3] !== 'up' || parts.length === 4) { // inject app_id and dev_id, but not for field topics
       payload.app_id = this.appId;
       payload.dev_id = devId;
     }
```

Comment:
100% agree for `app_id`, 95% agree for `dev_id` for several reasons:

- In the Node-RED Node we need to inject it anyway since we can only output a single object
- I think both or at least `dev_id` should be in the MQTT payload anyway because most users will listen for messages from all devices and it doesn't make sense to have them extract it from the topic if we can just include it.